File: minseon/22_12_07_01.py
# 연속된 수의 합

#1차시도

def solution(num, total):
    answer = []
    middlenum = total//num
    if num%2 == 0:
        middlenum2 = middlenum+1
        answer.append(middlenum)
        answer.append(middlenum2)
        min = middlenum - 1
        max = middlenum2 + 1
        answer.append(min)
        answer.append(max)

        print(f'배열의 합:{sum(answer)}')
        #for문을 돌리던가 while문을 돌려서 배열의 합이 total과 같아질 때 까지 max에 +1씩 하고..min에 -1씩 하면 될 것 같은데..
        while sum(answer) < total :
            min -=1
            max += 1
            answer.append(min)
            answer.append(max)
            print(f'num이 짝수일 때 min의 값 : {min} , num이 짝수일 때 max의 값 : {max}')
        answer.sort()
        print(f'짝수일때 answer : {answer}')
    else:
        answer.append(middlenum)
        min = middlenum - 1
        max = middlenum + 1
        answer.append(min)
        answer.append(max)

        print(f'배열의 합:{sum(answer)}')
        while sum(answer) < total :
            min -= 1
            max += 1
            answer.append(min)
            answer.append(max)
            print(f'num이 짝수일 때 min의 값 : {min} , num이 짝수일 때 max의 값 : {max}')
        answer.sort()
        print(f'홀수일때 answer : {answer}')

    return answer
    
    #실패..!

File: sein/22_12_07_01.java
//연속된 세 개의 정수를 더해 12가 되는 경우는 3, 4, 5입니다. 
//두 정수 num과 total이 주어집니다. 연속된 수 num개를 더한 값이 total이 될 때, 정수 배열을 오름차순으로 담아 return하도록 solution함수를 완성해보세요.

class Solution {
    public int[] solution(int num, int total) {
        int[] result = new int[num];
        int count = 0;
        int sum = 0;
        
        
        for(int i=(-num); i<=1000; i++){
            result[count] = i;
            sum += i;
            
            count++;
            if(count==num){
                if(sum==total){
                    break;
                }else{
                    i = result[0];
                    count = 0;
                    sum=0;
                }
            }
        }
        
        return result;
    }
}
